sw/device/lib/testing/i2c_testutils.c

```diff
@@ -91,46 +91,72 @@ static const i2c_pinmux_pins_t kI2cPinmuxPins[] = {
 };
 
 /**
- * This table store i2c pin mappings of different platforms.
- * This is used to connect i2c instances to mio pins based on the platform.
+ * Map the combination of platform and I2C ID to pins for SCL and SDA.
  */
-static const i2c_platform_pins_t kI2cPlatformPins[] = {
-    // Hyper310 bitstream.
-    [I2cPinmuxPlatformIdHyper310] =
-        {.sda =
-             {
-                 .mio_out = kTopEarlgreyPinmuxMioOutIoa7,
-                 .insel = kTopEarlgreyPinmuxInselIoa7,
-             },
-         .scl =
-             {
-                 .mio_out = kTopEarlgreyPinmuxMioOutIoa8,
-                 .insel = kTopEarlgreyPinmuxInselIoa8,
-             }},
-    // DV.
-    [I2cPinmuxPlatformIdDvsim] = {.sda =
-                                      {
-                                          .mio_out =
-                                              kTopEarlgreyPinmuxMioOutIob10,
-                                          .insel = kTopEarlgreyPinmuxInselIob10,
-                                      },
-                                  .scl =
-                                      {
-                                          .mio_out =
-                                              kTopEarlgreyPinmuxMioOutIob9,
-                                          .insel = kTopEarlgreyPinmuxInselIob9,
-                                      }},
-    // Cw310 PMOD.
-    [I2cPinmuxPlatformIdCw310Pmod] = {
-        .sda =
-            {
-                .mio_out = kTopEarlgreyPinmuxMioOutIob12,
-                .insel = kTopEarlgreyPinmuxInselIob12,
-            },
-        .scl = {
-            .mio_out = kTopEarlgreyPinmuxMioOutIob11,
-            .insel = kTopEarlgreyPinmuxInselIob11,
-        }}};
+static status_t map_platform_to_pins(i2c_pinmux_platform_id_t platform,
+                                     uint8_t i2c_id,
+                                     i2c_platform_pins_t *pins) {
+  TRY_CHECK(pins != NULL);
+  switch (platform) {
+    case I2cPinmuxPlatformIdHyper310:  // CW310 HyperDebug
+      *pins =
+          (i2c_platform_pins_t){.sda =
+                                    {
+                                        .mio_out = kTopEarlgreyPinmuxMioOutIoa7,
+                                        .insel = kTopEarlgreyPinmuxInselIoa7,
+                                    },
+                                .scl = {
+                                    .mio_out = kTopEarlgreyPinmuxMioOutIoa8,
+                                    .insel = kTopEarlgreyPinmuxInselIoa8,
+                                }};
+      break;
+    case I2cPinmuxPlatformIdDvsim:  // DV
+      // In DV, there's one agent for each I2C instance, with a fixed set of
+      // muxed pins.
+      switch (i2c_id) {
+        case 0:  // I2C0 uses the same pins as CW310 HyperDebug
+          TRY(map_platform_to_pins(I2cPinmuxPlatformIdHyper310, i2c_id, pins));
```

Comment:
I feel a little iffy about tying the platforms together like this, since DV could evolve into something different while hyper310 could not change. However, no need to change it on my account if you don't feel similarly.

Reply:
I see your point -- but this intertwining existed in the code already before this commit implicitly, this commit just makes it explicit. If the pins change in DV, this will no longer hold and tests will fail. I think the 'perfect' solution would be to describe the DV environment in an Hjson and generate RTL and headers from it, but in the meantime I think this is as good a solution as we can get (without duplicating code).